falco.yaml

```
@@ -309,3 +309,10 @@ metadata_download:
   max_mb: 100
   chunk_wait_us: 1000
   watch_freq_sec: 1
+
+# This is the dimension that a single buffer in our drivers will have. (BPF, kmod, modern BPF)
+# Please note:
+# - This number is expressed in bytes.
+# - This number must be a multiple of your system page size, otherwise the allocation will fail. 
+# - If you leave `0`, every driver will set its internal default dimension.
+single_buffer_dimension: 0
```

```suggestion
syscall_buffer_size: 0
```

Since this option affects the `syscall` data source only, I'd prefix it with `syscall_` and move it close to other `syscall_` options.
Also, I'd simplify the name and explain in the comment what that size means and how it affects the operations (eg. the driver will create a buffer per CPU, `syscall_buffer_size` specifies the size in bytes of each buffer...)

---

completely agree with the new name!

---

If we decide to have this param be just a multiple of page size need to adjust the `_size` at the end and explain.

[nit] More additional explanations for end users less familiar with architecture. Something like:

Falco uses a shared buffer between kernel and userspace to serve events up to userspace for subsequent processing and rules matching. With the `TBD_BUFFER_PARAM_NAME` you can increase or decrease the default buffer size for either driver (BPF, kmod, modern BPF) wrt the system call tracing functionality. In more detail, Falco uses one buffer for every online CPU, but you generically adjust the size once with this parameter. Increasing the buffer can help reducing kernel side syscall drops while decreasing the buffer can help speed up the entire engine.

---

One more thought that just occurred to me, maybe have default value be actual default value and not 0 to be consistent with other parameters? @Andreagit97 @leogr